compiler/rustc_resolve/src/lib.rs

```diff
@@ -1679,6 +1680,13 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
     }
 
     pub fn into_outputs(self) -> ResolverOutputs {
+        let visibilities_hash = {
+            let mut hasher = StableHasher::new();
+            let mut hcx = self.create_stable_hashing_context();
+            self.visibilities_for_hashing.hash_stable(&mut hcx, &mut hasher);
```

Comment:
I thought about doing this operation in `fn feed_visibility` and keeping only the hash in `Resolver` instead of the whole `visibilities_for_hashing` table.

Is creating the hashing context a more or less expensive operation than adding a table element?

Reply:
Creating a hashing context is very cheap. It's just moving some pointers around. I guess we could keep a `StableHasher` around, feed it the individual items, and finish into a hash here.